File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // Keep two pointers, moving from the LHS, one on the RHS
        // If the sum is greater, move Right pointer --
        // If the sum is lesser, move the left pointer ++; 


        int lower = 0; 
        int right = numbers.length - 1; 
        int sum = 0; 
        int [] ans = new int[2]; 
        while (lower < right) {
            sum = numbers[lower] + numbers[right]; 
            if (sum == target) {
                ans[0] = ++lower; 
                ans[1] = ++right;  
                break; 
            }
            
            else if (sum > target) {
                right--;
            }
            else if (sum < target) {
                lower++; 
            }
            
        
        }
        return ans; 
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [-1] * n
        def counts(curr):
            if curr > n:
                return 0
            if curr == n:
                return 1
                
            if dp[curr] == -1:
                ans = counts(curr+1) + counts(curr+2)
                dp[curr] = ans
                return ans
            else:
                return dp[curr]
            
        
        return counts(0)